Mage/src/main/java/mage/abilities/costs/mana/ManaCostsImpl.java

```diff
@@ -344,6 +348,22 @@ public void assignPayment(Game game, Ability ability, ManaPool pool, Cost costTo
         }
         // stop using mana of the clicked mana type
         pool.lockManaType();
+        handleForcedToPayOnlyForCurrentPayment(game, pool, referenceCosts);
+    }
+
+    private void handleForcedToPayOnlyForCurrentPayment(Game game, ManaPool pool, ManaCosts referenceCosts) {
+        // for Word of Command
+        if (pool.isForcedToPay()) {
+            if (referenceCosts != null && this.getPayment().equals(referenceCosts.getPayment())) {
```

Comment:
Am I reading this correctly: if referenceCosts are not null and the current payment is equal to the referenceCosts payment --> rollback.

I thought for word of command we wanted the opposite? That is, if the current payment is **not** equal to reference cost of the commanded card, rollback (since the player produced mana that does not directly pay for the commanded card, which is illegal when casting of Word of Command).

I'm new to Java so I could just be missing something here!

Reply:
> I thought for word of command we wanted the opposite?

The thing is though that handleForcedToPayOnlyForCurrentPayment not doing anything is the desired outcome - it means that at least some of the drawn mana was assigned to the payment (and therefore the resulting payment is different from the one in the referenceCosts snapshot). HandleForcedToPayOnlyForCurrentPayment is only supposed to go off on a "wrong" payment - one that only drew mana to the pool and none to the pending payment.